test/app.go

```diff
@@ -253,6 +261,12 @@ func (s *testSettings) apply() {
 		cache.ResetThemeCaches()
 		intapp.ApplySettings(s, s.app)
 		s.app.propertyLock.Unlock()
+
+		s.listenersLock.Lock()
+		defer s.listenersLock.Unlock()
+		for _, l := range s.listeners {
+			l(s)
+		}
```

Comment:
Hmm. This read is using listenersLock and the append is using propertyLock. That does not sound ideal. I suppose both ought to be the latter and then we don't need a new lock?